File: June 2025/11.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findLength(vector<int> &color, vector<int> &radius) {
        stack<pair<int, int>> st;  // Stack to track remaining balls (color, radius)

        for (size_t i = 0; i < color.size(); i++) {
            // If stack is not empty and top element matches current color & radius, remove it
            if (!st.empty() && st.top() == make_pair(color[i], radius[i])) {
                st.pop();
            } else {
                st.push(make_pair(color[i], radius[i]));
            }
        }

        return st.size();  // Remaining balls count
    }
};

File: June 2025/11.py
# User function Template for python3
class Solution:
    def findLength(self, color, radius):
        stack = []

        for i in range(len(color)):
            # If stack is not empty and the top element has the same color & radius, pop it
            if stack and stack[-1] == (color[i], radius[i]):
                stack.pop()
            else:
                stack.append((color[i], radius[i]))

        return len(stack)
